Adds the Smoking Room icebox surface ruin

Rhials opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

My contribution to the March Into Mapping/Mapness event -- The forlorn homestead of mister Charles Morlbaro, a high-brow individual with a taste for cigarettes. This ruin spawns on the surface level of Icebox.

![image](https://user-images.githubusercontent.com/28870487/224414415-6bacf208-9a39-43e4-8f79-60d142049a17.png)

A local of the ice planet, Morlbaro lived a peaceful life at home, until an _incident_ compromised one of his windows. Fearing the cold and unable to keep anything larger than a cigarette lit, he sequestered himself in his private smoking room, and hasn't been seen since.

...

Oh, right, the loot. That's the most important part for some of you guys.

![image](https://user-images.githubusercontent.com/28870487/224161740-cc25f84e-3aaf-4f78-9ea9-06c79e376cdb.png)

Where there's smoke, there's fire, and where there's cigarettes, there's lighters. Mister Morlbaro collected novelty lighters, which have to be worth something to someone, right? You might be able to find some if you dig around in his belongings. Just try not to disturb him if you choose to enter his smoking room.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

### Mapping March

<!-- If your pull request is part of Mapping March and you want to earn an antagonist token for your FIRST mapping pull request submitted this month, please include your ckey. For more information, read about the event here: https://hackmd.io/@EOBGames/BkwuRlxkh -->

Ckey to recieve rewards: theduffelbag

## Why It's Good For The Game

Ruin variety is nice. This one doesn't have any particularly spectacular loot, but it's a surface ruin.

Bro space ruins are sooooo 2020 bro, it's all about icebox ruins now bro.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Rhials
add: Adds the Smoking Room icebox ruin, found on the station level!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rhials committed:
adds some lighters for the smoking room

I have a very old branch to revisit and steal code from brb

Rhials committed:
adds spritework, more cool lighter stuff

Rhials committed:
initial mapping, areas

Rhials committed:
more adjustments, adds the lighter spawner

Rhials committed:
commit

Rhials committed:
furhter progress

Rhials committed:
removes all that fucking grime

Rhials committed:
this map is LEAKING

Rhials committed:
fixes the clear lighter sprite

Rhials committed:
fixes the active turf issue

Rhials committed:
oh my goodness I think the map is ready!

Rhials committed:
legally distinct!

Rhials committed:
last touches

VladinXXV commented:
can i convince you to throw the slime zippo in there too? i think it's pretty neat. good ruin overall.

MrMelbert reviewed (changes requested):
You do not appear to have the githooks setup, as your map does not have the header
`//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE` (that is to say it wasn't converted to TGM format)

Fikou commented on code/game/objects/items/cigs_lighters.dm:
```diff
@@ -916,6 +916,60 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+	fancy = FALSE
+
+/obj/item/lighter/bright/examine(mob/user)
+	..()
```
you need to include return value

ArcaneMusic commented on code/game/objects/items/cigs_lighters.dm:
```diff
@@ -916,6 +916,60 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+/obj/item/lighter/skull
+	name = "badass zippo"
+	desc = "An absolutely badass zippo lighter. Just look at that skull!"
+	custom_price = PAYCHECK_CREW * 6
```
This doesn't give it a sale value, you'll need to make a cargo export datum for it to be worth that much in cargo.
Check the `cargo/exports` folder for examples.

Rhials committed:
god I sure hope this is what fikou meant

Rhials committed:
implements some of melberts advice

removes all of those stupid freezing floors and adds some cool wood trims for the new carpeting

Loafin34 commented:
If only I knew modsuit code, So I could make a lighter module... alas.

Maurukas reviewed (changes requested):
The ruin itself looks good, however you're saving in .dmm format. You need to convert the map to tgm format. You can do this by setting up the git hooks, or by opening the map in StrongDMM2 and going to File -> Preferences and changing the Save Format to TGM (then saving).

Mark ready for review when you're done, or we can help in mapping-general on the discord.

Rhials committed:
lets wrap this up

adds export list for rare lighters, adds gassy remains

Rhials committed:
converts the file to tgm

puts the smokey remains in place and fixes some wood trims

Rhials committed:
final adjustments

Maurukas reviewed (approved):
Maps look good. However, the smoke effect on the corpse is runtiming for me.

[18:30:14] Runtime in remains.dm,42: Cannot read /datum/reagent/consumable/nutr... (/datum/reagent/consumable/nutraslop).name
  proc name: release smoke (/obj/effect/decal/remains/human/smokey/proc/release_smoke)
  usr: Maurukas/(Vire-Sartri)
  usr.loc: (Smoking Room (226,211,4))
  src: the remains (/obj/effect/decal/remains/human/smokey)
  src.loc: the carpet (226,211,4) (/turf/open/floor/carpet/blue)

I'm not immediately sure why it's doing this, I would bother a coder.

Rhials committed:
its a typepath so I cant use .name what was I THIN

LemonInTheDark commented on code/game/objects/effects/decals/remains.dm:
```diff
@@ -39,7 +39,7 @@
 
 /obj/effect/decal/remains/human/smokey/proc/release_smoke(mob/living/smoke_releaser)
-	visible_message(span_warning("[smoke_releaser] disturbs the [src], which releases a huge cloud of [that_shit_that_killed_saddam.name]!"))
+	visible_message(span_warning("[smoke_releaser] disturbs the [src], which releases a huge cloud of gas!"))
```
worth noting you coulda used initial(that_shit_that_killed_sadam.name), it's how you access non list vars off typepaths
